mongodbatlas/resource_mongodbatlas_org_invitation.go

```diff
@@ -0,0 +1,269 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+func resourceMongoDBAtlasOrgInvitation() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceMongoDBAtlasOrgInvitationCreate,
+		ReadContext:   resourceMongoDBAtlasOrgInvitationRead,
+		DeleteContext: resourceMongoDBAtlasOrgInvitationDelete,
+		UpdateContext: resourceMongoDBAtlasOrgInvitationUpdate,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMongoDBAtlasOrgInvitationImportState,
+		},
+		Schema: map[string]*schema.Schema{
+			"org_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"username": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"invitation_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"expires_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"roles": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceMongoDBAtlasOrgInvitationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// Get client connection.
+	conn := meta.(*MongoDBClient).Atlas
+	ids := decodeStateID(d.Id())
+	orgID := ids["org_id"]
+	username := ids["username"]
+	invitationID := ids["invitation_id"]
+
+	orgInvitation, resp, err := conn.Organizations.Invitation(context.Background(), orgID, invitationID)
+	if err != nil {
+		// case 404
+		// deleted in the backend case
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
```

Comment:
[not-blocking] I know this is the common pattern in the code base but since we are on a more recent version of go we can instead do
```suggestion
		var target *atlas.ErrorResponse
		if errors.As(err, &target) && target.ErrorCode == "NOT_FOUND" {
			d.SetId("")
			return nil
		}
```

again not blocking and one of my planned things to refactor at some point to improve our error handling

Reply:
Fixed